include/MySQL_HostGroups_Manager.h

```diff
@@ -30,7 +30,7 @@
 #define MYHGM_MYSQL_SERVERS "CREATE TABLE mysql_servers ( hostgroup_id INT NOT NULL DEFAULT 0 , hostname VARCHAR NOT NULL , port INT NOT NULL DEFAULT 3306 , gtid_port INT NOT NULL DEFAULT 0 , weight INT NOT NULL DEFAULT 1 , status INT NOT NULL DEFAULT 0 , compression INT NOT NULL DEFAULT 0 , max_connections INT NOT NULL DEFAULT 1000 , max_replication_lag INT NOT NULL DEFAULT 0 , use_ssl INT NOT NULL DEFAULT 0 , max_latency_ms INT UNSIGNED NOT NULL DEFAULT 0 , comment VARCHAR NOT NULL DEFAULT '' , mem_pointer INT NOT NULL DEFAULT 0 , PRIMARY KEY (hostgroup_id, hostname, port) )"
 #define MYHGM_MYSQL_SERVERS_INCOMING "CREATE TABLE mysql_servers_incoming ( hostgroup_id INT NOT NULL DEFAULT 0 , hostname VARCHAR NOT NULL , port INT NOT NULL DEFAULT 3306 , gtid_port INT NOT NULL DEFAULT 0 , weight INT NOT NULL DEFAULT 1 , status INT NOT NULL DEFAULT 0 , compression INT NOT NULL DEFAULT 0 , max_connections INT NOT NULL DEFAULT 1000 , max_replication_lag INT NOT NULL DEFAULT 0 , use_ssl INT NOT NULL DEFAULT 0 , max_latency_ms INT UNSIGNED NOT NULL DEFAULT 0 , comment VARCHAR NOT NULL DEFAULT '' , PRIMARY KEY (hostgroup_id, hostname, port))"
 #endif /* DEBUG */
-#define MYHGM_MYSQL_REPLICATION_HOSTGROUPS "CREATE TABLE mysql_replication_hostgroups (writer_hostgroup INT CHECK (writer_hostgroup>=0) NOT NULL PRIMARY KEY , reader_hostgroup INT NOT NULL CHECK (reader_hostgroup<>writer_hostgroup AND reader_hostgroup>=0) , check_type VARCHAR CHECK (LOWER(check_type) IN ('read_only','innodb_read_only','super_read_only')) NOT NULL DEFAULT 'read_only' , comment VARCHAR NOT NULL DEFAULT '' , UNIQUE (reader_hostgroup))"
+#define MYHGM_MYSQL_REPLICATION_HOSTGROUPS "CREATE TABLE mysql_replication_hostgroups (writer_hostgroup INT CHECK (writer_hostgroup>=0) NOT NULL PRIMARY KEY , reader_hostgroup INT NOT NULL CHECK (reader_hostgroup<>writer_hostgroup AND reader_hostgroup>=0) , check_type VARCHAR CHECK (LOWER(check_type) IN ('read_only','innodb_read_only','super_read_only','read_only|innodb_read_only','read_only&innodb_read_only')) NOT NULL DEFAULT 'read_only' , comment VARCHAR NOT NULL DEFAULT '' , UNIQUE (reader_hostgroup))"
```

Comment:
Though I cannot specifically foresee why someone would need an `read_only&innodb_read_only` . The database is definitely not writable if even one of these is set, so that's what we should go for

Reply:
We thought about this for a while.
We didn't foresee someone wanting to use `or` but now we realize it makes sense.
For `and`, we can find an use case as well.

For example, say you have 3 sets of servers:
a) the writer : read_only=0
b) the readers in the same DC : `read_only=1` 
c) the readers in a remote DC that you don't want to normally use `read_only=1&innodb_read_only=1`

You could configure proxysql as:
HG1 / HG2 , check on `read_only`
HG3 / HG4 , check on `read_only&innodb_read_only`

With this setup, you could send writes to HG1 and reads to HG3 (the local DC).
Keep in mind that `mysql_replication_hostgroups` is replication agnostic, and that the same servers can be in multiple hostgroups. Therefore even if HG4 is not replicating from HG3, proxysql won't care about replication and place servers in the right HG based on the check.

Reply:
I see. This is an interesting use case you bring up. Actually, it might help solve some of our problems. Maybe your example is slightly wrong because in Aurora, the "local read replica"s use `innodb_read_only`. So you probably meant that readers in the same DC have `read_only=1`.

Suppose we have this particular setup:
[Aurora Master 1] ------> [Aurora Replica 2]
           |
           |
           v
[Aurora "Cross Region" Replica 3] --------> [ Aurora "Cross Region" Replica 4]

In this setup, we have noted that:
Server 1: innodb_read_only=0, read_only=0
Server 2: innodb_read_only=1, read_only=0
Server 3: innodb_read_only=0, read_only=1 [we set read_only via parameter groups]
Server 4: innodb_read_only=1, read_only=1 [we set read_only via parameter groups]

In our case, we don't support writing from multiple regions, so we switch the `read_only` flag to `True` on the "Cross Region" replicas.

The innodb_read_only&read_only flag will let us target `Server 4` specifically if we have to in the "remote region". 

"Aurora Global DB" does the flags slightly differently, I haven't looked into it deeply.